ui/packages/catalog-ui-search/src/main/webapp/react-component/utils/validation/validation.tsx

```diff
@@ -38,31 +51,275 @@ export function showErrorMessages(errors: any) {
+    errors.add({
+      title: 'Invalid geometry filter',
+      body: err,
+    })
+  })
+  return Array.from(errors)
+}
+
+export function validateGeo(
+  key: string,
+  value: string,
+  value1?: any,
+  value2?: any,
+  value3?: any
+) {
+  switch (key) {
+    case 'lat':
+    case 'north':
+    case 'south':
+      return validateDDLatLon('latitude', 90, value)
+    case 'lon':
+    case 'west':
+    case 'east':
+      return validateDDLatLon('longitude', 180, value)
+    case 'dmsLat':
+    case 'dmsNorth':
+    case 'dmsSouth':
+      return validateDmsLatLon('latitude', value)
+    case 'dmsLon':
+    case 'dmsEast':
+    case 'dmsWest':
+      return validateDmsLatLon('longitude', value)
+    case 'usng':
+      return validateUsng(value)
+    case 'utmUpsEasting':
+    case 'utmUpsNorthing':
+    case 'utmUpsZone':
+    case 'utmUpsHemisphere':
+      return validateUtmUps(key, value, value1, value2, value3)
+    case 'radius':
+    case 'lineWidth':
+      return validateRadiusLineBuffer(key, value)
+    default:
+  }
+}
+
+export function getErrorComponent(errorState: ErrorState) {
+  return errorState.error ? (
+    <Invalid>
+      <WarningIcon className="fa fa-warning" />
+      <span>{errorState.message}</span>
+    </Invalid>
+  ) : null
+}
+
+function getGeometryErrors(filter: any): Set<string> {
+  const geometry = filter.geojson && filter.geojson.geometry
+  const bufferWidth =
+    filter.geojson.properties.buffer && filter.geojson.properties.buffer.width
+  const errors = new Set<string>()
+  if (!geometry) {
+    return errors
+  }
+  switch (filter.geojson.properties.type) {
+    case 'Polygon':
+      if (geometry.coordinates[0].length < 4) {
+        errors.add(
+          'Polygon coordinates must be in the form [[x,y],[x,y],[x,y],[x,y], ... ]'
+        )
+      }
+      break
+    case 'LineString':
+      if (geometry.coordinates.length < 2) {
+        errors.add('Line coordinates must be in the form [[x,y],[x,y], ... ]')
+      }
+      if (!bufferWidth || bufferWidth == 0) {
+        errors.add('Line buffer width must be greater than 0.000001')
+      }
+      break
+    case 'Point':
+      if (!bufferWidth || bufferWidth < 0.000001) {
+        errors.add('Radius must be greater than 0.000001')
+      }
+      if (
+        geometry.coordinates.some(
+          (coord: any) => !coord || coord.toString().length == 0
+        )
+      ) {
+        errors.add('Coordinates must not be empty')
+      }
+      break
+    case 'BoundingBox':
+      const box = filter.geojson.properties
+      if (!box.east || !box.west || !box.north || !box.south) {
+        errors.add('Bounding Box must have valid values')
+      }
+      break
   }
   return errors
 }
+
+function validateDDLatLon(label: string, defaultCoord: number, value: string) {
+  let message = ''
+  let defaultValue
+  if (value !== undefined && value.length === 0) {
+    message = `${label.replace(/^\w/, c => c.toUpperCase())} cannot be empty`
```

Comment:
I'd pull this into a function like we talked about

Reply:
got it